packages/odyssey-react/src/components/Checkbox/index.tsx

```diff
@@ -10,164 +10,5 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { useCallback, useRef, useEffect } from "react";
-import type {
-  FunctionComponent,
-  FocusEventHandler,
-  ChangeEvent,
-  RefCallback,
-} from "react";
-import { Check, Minus } from "../Icon";
-import styles from "./Checkbox.module.scss";
-import { useOid } from "../../utils";
-
-export type Props = {
-  /**
-   * The underlying input element id attribute. Automatically generated if not provided
-   */
-  id?: string;
-
-  /**
-   * The form field label
-   */
-  label: string;
-
-  /**
-   * Callback to provide a reference to the underlying input element
-   * @param {Object} instance the input element or null
-   */
-  inputRef?: RefCallback<HTMLInputElement>;
-
-  /**
-   * The underlying input element name attribute
-   */
-  name: string;
-
-  /**
-   * The underlying input element required attribute
-   * @default true
-   */
-  required?: boolean;
-
-  /**
-   * The underlying input element disabled attribute
-   * @default false
-   */
-  disabled?: boolean;
-
-  /**
-   * The underlying input element value attribute
-   */
-  value: string;
-
-  /**
-   * The input element checked state for controlled components
-   * @default false
-   */
-  checked?: boolean;
-
-  /**
-   * The input element indeterminate state for controlled components
-   * @default false
-   */
-  indeterminate?: boolean;
-
-  /**
-   * Callback executed when the input fires a blur event
-   * @param {Object} event the event object
-   */
-  onBlur?: FocusEventHandler<HTMLInputElement>;
-
-  /**
-   * Callback executed when the input fires a change event
-   * @param {Object} event the event object
-   * @param {string} value the string value of the input
-   */
-  onChange?: (event: ChangeEvent<HTMLInputElement>, value: string) => void;
-
-  /**
-   * Callback executed when the input fires a focus event
-   * @param {Object} event the event object
-   */
-  onFocus?: FocusEventHandler<HTMLInputElement>;
-};
-
-/**
- * A clickable Checkbox used for form submissions and most in-page interactions.
- *
- * @component
- * @example
- * <Checkbox label="Checkbox label" name="checkbox_name" value="checkbox_value" />
- */
-const Checkbox: FunctionComponent<Props> = (props) => {
-  const {
-    disabled = false,
-    id,
-    inputRef,
-    label,
-    checked = false,
-    name,
-    onBlur,
-    onChange,
-    onFocus,
-    required = true,
-    value,
-    indeterminate = false,
-  } = props;
-
-  const oid = useOid(id);
-
-  const internalRef = useRef<HTMLInputElement>(null);
-  useEffect(() => {
-    if (inputRef) {
-      inputRef(internalRef.current);
-    }
-  }, [internalRef, inputRef]);
-
-  useEffect(() => {
-    if (internalRef.current) {
-      if (indeterminate) {
-        internalRef.current.indeterminate = true;
-      } else {
-        internalRef.current.indeterminate = false;
-      }
-    }
-  }, [indeterminate, internalRef]);
```

Comment:
Why was this code removed? Without it the checkbox accessibility is affected. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role

Reply:
@andrewberg-okta I removed it because I saw an opportunity to simplify the implementation by driving the `intermediate` styles via a class rather than a property on the element itself, assigned via an intermediate ref we have to manage. I also [added a new story for this use case you can see here](https://eea4053.ods.dev/sb/index.html?path=/story/components-checkbox--indeterminate) but will look into a unit test as well. Is there something about this property that requires it to be assigned to the DOM node in this fashion but doesn't allow it to be passed as a prop to input normally?

Reply:
ah I see, `indeterminate` does not have an HTML api only JS. I can rework and add a test - good feedback

Reply:
@andrewberg-okta updated! see https://github.com/okta/odyssey/pull/1082/files#diff-290b338a454d8d9b60062d451e871f3e576149b433d77aaa53f6a48ad9d6efffR83-R86